api/provider/aws/deploy_create_test.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/api/tag"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeploy_createTags(t *testing.T) {
+	tagStore := tag.NewMemoryStore()
+	deploy := NewDeployProvider(nil, tagStore, nil)
+	arn := "deploy_arn"
+
+	model := &models.Deploy{
+		DeployID:   "deploy_id",
+		Version:    "deploy_version",
+		DeployName: "deploy_name",
+	}
+
+	if err := deploy.createTags(model, arn); err != nil {
+		t.Fatal(err)
+	}
+
+	expectedTags := models.Tags{
+		{
+			EntityID:   "deploy_id",
+			EntityType: "deploy",
+			Key:        "name",
+			Value:      "deploy_name",
+		},
+		{
+			EntityID:   "deploy_id",
+			EntityType: "deploy",
+			Key:        "version",
+			Value:      "deploy_version",
+		},
+		{
+			EntityID:   "deploy_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "deploy_arn",
+		},
+	}
+
+	for _, tag := range expectedTags {
+		assert.Contains(t, tagStore.Tags(), tag)
+	}
+}
+
+func TestDeploy_renderTaskDefinition_InvalidRequest(t *testing.T) {
+	deploy := NewDeployProvider(nil, nil, nil)
```

Comment:
Can we switch these to use a table test?
```
func TestDeploy_renderTaskDefinitionErrors(t *testing.T){
    cases := map[string]*ecs.TaskDefinition{
        "Custom Family Name": &ecs.TaskDefinition{...},
        "No Container Definitions": &ecs.TaskDefinition{},
    }
 
   ...
}
```

Reply:
Ahh good call, will do

Reply:
✅